slide_analysis/UI/view/image_helper.py

```diff
@@ -1,34 +1,40 @@
 import openslide
+import time
 from PIL import ImageQt
+from PyQt5.QtCore import QRect
 
 from slide_analysis.constants.tile import BASE_TILE_WIDTH, BASE_TILE_HEIGHT
 from slide_analysis.UI.view.constants import BASE_SCALE_FACTOR, SCALE_MULTIPLIER
 
 
 class ImageHelper:
-    def __init__(self, filepath):
-        self.filepath = filepath
-        self.openslide_image = openslide.OpenSlide(filepath)
-        self.current_level = self.openslide_image.level_count - 1
+    def __init__(self, filename):
+        self.filename = filename
+        self.openslide_image = openslide.OpenSlide(filename)
+        self.max_level = self.openslide_image.level_count - 1
+        self.current_level = self.max_level
         self.level_dimensions = self.openslide_image.level_dimensions
-        self.current_coordinates = (0, 0)
+        self.current_displayed_image_part_coordinates = (0, 0)
+        self.current_image_part_coordinates = self.current_displayed_image_part_coordinates
+
+        self.current_displayed_image_part_size = self.level_dimensions[self.current_level]
+        self.current_image_part_size = self.current_displayed_image_part_size
+
         self.image_dimensions = self.level_dimensions[0]
         self.scale_factor = BASE_SCALE_FACTOR
         self.current_movement_step = (self.image_dimensions[0] // self.scale_factor,
                                       self.image_dimensions[1] // self.scale_factor)
-        self.current_window_size = self.level_dimensions[self.current_level]
-        self.image = self.openslide_image.read_region(self.current_coordinates, self.current_level,
+        self.image = self.openslide_image.read_region(self.current_displayed_image_part_coordinates, self.current_level,
```

Comment:
change current_displayed_image_part_coordinates to current_image_part_coordinates